macros/src/main/scala/enumeratum/ValueEnumMacros.scala

```diff
@@ -217,6 +217,14 @@ object ValueEnumMacros {
                       c.enclosingPosition,
                       s"${declTree.symbol} has a value with the wrong type: $i:${i.getClass}, instead of ${classTag.runtimeClass}"
                     )
+
+                  case (_, AssignOrNamedArg(Ident(`valueTerm`), expr)) =>
```

Comment:
I think the other cases could be refactored using `expr` en `eval` and remove some code.

Reply:
Agreed. I think it might be able to get rid of the special handling per value type. Likely best to do it in another PR though :)